storage_service/common/management/commands/import_aip.py

```diff
@@ -0,0 +1,405 @@
+import shlex
+import shutil
+import subprocess
+import tarfile
+import tempfile
+
+import bagit
+from django.core.management.base import BaseCommand
+from django.db.utils import IntegrityError
+
+from administration.models import Settings
+from common import utils
+from locations import models
+
+
+# Suppress the logging from models/package.py
+logging.config.dictConfig({
+    'version': 1,
+    'disable_existing_loggers': True,
+})
+
+
+DEFAULT_AS_LOCATION = 'default_AS_location'
+DEFAULT_UNIX_OWNER = 'archivematica'
+ANSI_HEADER = '\033[95m'
+ANSI_OKGREEN = '\033[92m'
+ANSI_WARNING = '\033[93m'
+ANSI_FAIL = '\033[91m'
+ANSI_ENDC = '\033[0m'
+
+
+class Command(BaseCommand):
+
+    help = 'Import an AIP into the Storage Service'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'aip_path', help='Full path to the AIP to be imported')
+        parser.add_argument(
+            '--aip-storage-location',
+            help='UUID of the AIP Storage Location where the imported AIP'
+                 ' should be stored. Defaults to default AS location.',
+            default=DEFAULT_AS_LOCATION,
+            required=False)
+        parser.add_argument(
+            '--pipeline',
+            help='UUID of a pipeline that should be listed as the AIP\'s'
+                 ' origin. Defaults to an arbitrary pipeline.',
+            required=False)
+        parser.add_argument(
+            '--decompress-source',
+            help='Use this flag to indicate that AIP_PATH should be'
+                 ' decompressed and the resulting DIRECTORY should be the'
+                 ' UNCOMPRESSED AIP to be imported.',
+            action='store_true',
+            default=False)
+        parser.add_argument(
+            '--compression-algorithm',
+            help='The compression algorithm to use when compressing the'
+                 ' imported AIP. Omit this if the AIP is already compressed or'
+                 ' if you want to import an uncompressed AIP as is.',
+            choices=utils.COMPRESSION_ALGORITHMS,
+            default=None)
+        parser.add_argument(
+            '--unix-owner',
+            help='The Unix system user that should own the file(s) of the'
+                 ' imported AIP. Default: {}'.format(DEFAULT_UNIX_OWNER),
+            default=DEFAULT_UNIX_OWNER)
+
+    def handle(self, *args, **options):
+        print(header(
+            'Attempting to import the AIP at {}.'.format(options['aip_path'])))
+        try:
+            import_aip(options['aip_path'],
+                       options['aip_storage_location'],
+                       options['decompress_source'],
+                       options['compression_algorithm'],
+                       options['pipeline'],
+                       options['unix_owner'])
+        except ImportAIPException as err:
+            print(fail(err))
+
+
+class ImportAIPException(Exception):
+    """An error occurred when attempting to import an AIP."""
```

Comment:
Should this subclass `CommandError`?

> If this exception is raised during the execution of a management command, it will be caught and turned into a nicely-printed error message to the appropriate output stream (i.e., stderr); as a result, raising this exception (with a sensible description of the error) is the preferred way to indicate that something has gone wrong in the execution of a command.